app/services/map_data_service.py

```diff
@@ -0,0 +1,288 @@
+
+@dataclass
+class Address:
+    postal_code: str
+    administrative_area_level_1: str
+    administrative_area_level_2: str
+    sublocality_level_1: str
+    formatted_address: str
+
+
+class GoogleMapService:
+    def __init__(self):
+        api_key = os.getenv('GOOGLE_MAPS_API_KEY')
+        if not api_key:
+            print("❌ Error: Set GOOGLE_MAPS_API_KEY environment variable")
+        self.client = googlemaps.Client(key=api_key)
+        self.target_address_types = [
+            "postal_code",
+            "administrative_area_level_1", 
+            "administrative_area_level_2",
+            "sublocality_level_1"
+        ]
+
+    def get_entire_address(self, place_id: str) -> Address:
+        try:
+            place = self.client.place(place_id=place_id)
+            return self._extract_address_components(place)
+        except Exception as e:
+            print(f"❌ Error: {e}")
+            return None
+
+    def get_address_components_as_list(self, place_id: str) -> List[Dict[str, Any]]:
+        try:
+            place = self.client.place(place_id=place_id)
+            return place.get("result", {}).get("address_components", [])
+        except Exception as e:
+            print(f"❌ Error: {e}")
+            return None
+
+    def get_formatted_address(self, place_id: str) -> str:
+        try:
+            place = self.client.place(place_id=place_id)
+            return place.get("result", {}).get("formatted_address", "")
+        except Exception as e:
+            print(f"❌ Error: {e}")
+            return None
+
+    def get_phone_number(self, place_id: str) -> str:
+        try:
+            place = self.client.place(place_id=place_id)
+            return place.get("result", {}).get("international_phone_number", "")
+        except Exception as e:
+            print(f"❌ Error: {e}")
+            return None
+
+    def get_opening_hours(self, current_opening_hours: Dict[str, Any]) -> OpeningHours:
+        today = datetime.now()
+        week_start = today - timedelta(days=today.weekday())
+        week_end = week_start + timedelta(days=6)
+        
+        week_start_str = week_start.strftime("%Y-%m-%d")
+        week_end_str = week_end.strftime("%Y-%m-%d")
+        last_refreshed = week_end.replace(hour=21, minute=0, second=0).isoformat() + "+03:00"
+        
+        weekly = {}
+        weekday_names = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"]
+        
+        periods = current_opening_hours.get("periods", [])
+        
+        for period in periods:
+            open_info = period.get("open", {})
+            close_info = period.get("close", {})
+            
+            day_name = weekday_names[open_info.get("day", 0)]
+            start_time = open_info.get("time", "12:00")
+            end_time = close_info.get("time", "21:00")
+            
+            start_formatted = f"{start_time[:2]}:{start_time[2:]}"
+            end_formatted = f"{end_time[:2]}:{end_time[2:]}"
```

Comment:
The time string slicing assumes that the times are always 4 digits (e.g., "0900"). Validate the format before slicing.